types/diags.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+)
+
+// TODO: Replace with diagnostics abstraction
+// Reference: https://github.com/hashicorp/terraform-plugin-framework/issues/24
+func diagsHasErrors(in []*tfprotov6.Diagnostic) bool {
```

Comment:
These have been implemented in a number of different packages now. Is it worth just pulling them into an `internal/diags` package at this point?

Reply:
🙁 that's what I was alluding to when I wanted to start working on diagnostics before this, but I didn't want to sway the design by doing so. I will refactor this common diagnostics functionality into an internal package as suggested.

Reply:
Updated in c417a51b79a1c88cd3fed1f3da16b39137e748e8